File: src/main/java/me/blake/pulsar/actions/impl/TitleAction.java
package me.blake.pulsar.actions.impl;

import lombok.AllArgsConstructor;
import me.blake.pulsar.actions.Action;
import me.blake.pulsar.messaging.Message;
import me.lucko.helper.text3.Text;
import org.bukkit.entity.Player;

@AllArgsConstructor
public class TitleAction extends Action {

    private String title;
    private String subtitle;
    private int fadeIn;
    private int stay;
    private int fadeOut;

    public static Action from(String rawAction) {
        String[] rawActionValues = rawAction.split(";");

        System.out.println("Raw Action Values: " + rawActionValues.length);
//        switch (rawActionValues.length) {
//            case
//        }
        return new TitleAction(rawAction, "", 0, 70, 0);
    }

    @Override
    public void execute(Player player, Message.Placeholder[] placeholders) {
        player.sendTitle(
                Text.colorize(new Message(title).transform(placeholders).firstString()),
                Text.colorize(new Message(subtitle).transform(placeholders).firstString()),
                fadeIn,
                stay,
                fadeOut
        );
    }

    @Override
    public void execute(Player player) {
        player.sendTitle(Text.colorize(title),
                subtitle,
                fadeIn,
                stay,
                fadeOut
        );
    }
}

File: src/main/kotlin/me/blake/pulsar/actions/impl/TitleAction.kt
package me.blake.pulsar.actions.impl

import me.blake.pulsar.actions.Action
import me.blake.pulsar.messaging.Message
import me.blake.pulsar.messaging.Message.Placeholder
import me.lucko.helper.text3.Text
import org.bukkit.entity.Player

class TitleAction(
    private val title: String = "",
    private val subtitle: String = "",
    private val fadeIn: Int = 0,
    private val stay: Int = 10,
    private val fadeOut: Int = 0
): Action() {
    override fun execute(player: Player, vararg placeholders: Placeholder) {
        player.sendTitle(
            Text.colorize(Message(title).transform(*placeholders).firstString()),
            Text.colorize(Message(subtitle).transform(*placeholders).firstString()),
            fadeIn,
            stay,
            fadeOut
        )
    }

    override fun execute(player: Player) {
        player.sendTitle(
            Text.colorize(title),
            subtitle,
            fadeIn,
            stay,
            fadeOut
        )
    }

    companion object Parser {
        fun parse(action: String): TitleAction {
            val split = action.split(";")
            return TitleAction(
                split[0],
                split.getOrElse(1) { "" },
                split.getOrElse(2) { 0 } as Int,
                split.getOrElse(3) { 10 } as Int,
                split.getOrElse(4) { 0 } as Int
            )
        }
    }
}